connector/datadogconnector/connector.go

```diff
@@ -81,7 +81,17 @@ func newTraceToMetricConnector(set component.TelemetrySettings, cfg component.Co
 	for _, val := range cfg.(*Config).Traces.ResourceAttributesAsContainerTags {
 		ctags[val] = ""
 	}
-	ddtags := attributes.ContainerTagFromAttributes(ctags)
+	ddTags := attributes.ContainerTagFromAttributes(ctags)
+
+	for key := range ctags {
+		// TODO: we need to expose the container mappings from the attributes package
+		tags := attributes.ContainerTagFromAttributes(map[string]string{key: ""})
+		// It means there is no mapping for the resource attribute
+		if len(tags) == 0 {
+			ddTags[key] = ""
+		}
```

Comment:
We have full control over the `attributes` package, we don't need to resort to hacks, we can just file a PR over there. I can merge this PR tomorrow morning if that PR has not landed, but I will wait until the PR is open